man/fi_msg.3.md

```diff
@@ -288,82 +288,112 @@ fi_sendmsg.
-struct fi_var_context contains an op_context field.  Applications may
+struct fi_recv_context contains a 'context' field.  Applications may
 modify this field prior to claiming the message.  When the claim
 operation completes, a standard receive completion entry will be
-generated on the completion queue.  The op_context of the associated
-CQ entry will be set to the op_context value passed in through
-the fi_var_context structure.
-
-Applications that do not wish to receive a variable message that they
-were notified of may discard it.  To discard a message, an application
-must post a receive operation with the FI_DISCARD flag set.  The
-receive context should be the struct fi_var_context from the
-notification.  When the FI_DISCARD flag is set, the receive input
-buffer(s) and length parameters are ignored.
+generated on the completion queue.  The 'context' of the associated
+CQ entry will be set to the 'context' value passed in through
+the fi_recv_context structure, and the CQ entry flags will have the
+FI_CLAIM bit set.
+
+Buffered receives that are not claimed must be discarded by the application
+when it is done processing the CQ entry data.  To discard a message, an
+application must post a receive operation with the FI_DISCARD flag set.
+The struct fi_recv_context returned as part of the notification must be
+provided as the receive operation's context. When the FI_DISCARD flag is set
+for a receive operation, the receive input buffer(s) and length parameters
+are ignored.
+
+IMPORTANT: Buffered receives must be claimed or discarded in a timely manner.
+Failure to do so may result in increased memory usage for network buffering
+or communication stalls.  Once a buffered receive has been claimed or
+discarded, the original CQ entry 'buf' or struct fi_recv_context data may no
+longer be accessed by the application.
 
 The use of the FI_CLAIM and FI_DISCARD operation flags is also
 described with respect to tagged message transfers in fi_tagged.3.
-Variable length tagged messages will include the message tag as part
-of the message notification.
-
-Support for variable messages is indicated through the FI_VARIABLE_MSG
-capability bit.  Additionally, the variable length message threshold
-may be obtained and/or adjusted using an endpoint's
-fi_getopt/fi_setopt operations.
+Buffered receives of tagged messages will include the message tag as part
+of the CQ entry, if available.
 
-The handling of variable message headers follows all message ordering
+The handling of buffered receives follows all message ordering
 restrictions assigned to and endpoint.  For example, completions
-may indicate the order in which variable messages arrived at the
-receiver.  However, the transfer of variable message data should be
-treated as conceptually occurring out of band.  No ordering within or
-between the data of variable messages is implied.
+may indicate the order in which received messages arrived at the
+receiver based on the endpoint attributes.
```

Comment:
Need to note that this eliminates the need to register receives that are smaller than the threshold size (when claiming)